ludwig/data/dataset/ray.py

```diff
@@ -100,8 +107,16 @@ def pipeline(
             shuffle: If true, the entire dataset is shuffled in memory before batching.
             fully_executed: If true, force full evaluation of the Ray Dataset by loading all blocks into memory.
             window_size_bytes: If not None, windowing is enabled and this parameter specifies the window size in bytes
-                    for the dataset.
+                    for the dataset. If None and the dataset is large, set to the window size determined at init.
         """
+        # If the user does not supply a window size and the dataset is large,
+        # set the window size to `<available memory> // 5`.
+        if self.auto_window and not window_size_bytes:
+            ds_memory_size = self.in_memory_size_bytes
+            cluster_memory_size = ray.cluster_resources()["memory"]
+            if cluster_memory_size < 5.0 * ds_memory_size:
+                window_size_bytes = cluster_memory_size // 5
```

Comment:
Would be good to add a logging statement letting the user know that we're setting a window size since this will improve throughput/reduce disk spillage at the cost of model performance.

Reply:
The updated logging statement should be added in here: [24f36f5](https://github.com/ludwig-ai/ludwig/pull/2721/commits/24f36f5829d2197a39ff847dc016fd2108341052)